### PR TITLE
RDM-6520: Validate the CaseRoles on the case submission

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseAccessService.java [hunk 1]
```diff
@@ -76,6 +78,11 @@ public AccessLevel getAccessLevel(ServiceAndUserDetails serviceAndUserDetails) {
         return userRoles;
     }
 
+    public Set<String> getCreateCaseRoles() {
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseAccessService.java]: this should to be renamed to a more generic `getCreateRoles` as it's also returning idam roles (which are not case roles)

File: src/main/java/uk/gov/hmcts/ccd/domain/service/createcase/AuthorisedCreateCaseOperation.java [hunk 2]
```diff
@@ -7,21 +7,23 @@
 import org.springframework.stereotype.Service;
 import uk.gov.hmcts.ccd.data.definition.CachedCaseDefinitionRepository;
 import uk.gov.hmcts.ccd.data.definition.CaseDefinitionRepository;
-import uk.gov.hmcts.ccd.data.user.CachedUserRepository;
-import uk.gov.hmcts.ccd.data.user.UserRepository;
 import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/createcase/AuthorisedCreateCaseOperation.java]: when testing this PR locally better making sure your user doesn't have already a [CREATOR] role assigned in the `case_users` table

File: src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseDetailsEndpointIT.java [hunk 3]
```diff
@@ -2705,7 +2708,11 @@ private void shouldReturn404WhenPostCreateCaseWithNoCreateFieldAccess(String use
     }
```
Comment (mario-paniccia) [inline src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseDetailsEndpointIT.java]: these test classes are becoming too big. 
Shall we introduce a integration test class dedicated to Case Roles?

File: src/main/java/uk/gov/hmcts/ccd/domain/service/createcase/AuthorisedCreateCaseOperation.java [hunk 4]
```diff
@@ -7,21 +7,23 @@
...
 import uk.gov.hmcts.ccd.endpoint.exceptions.ValidationException;
 
 import java.util.HashMap;
 import java.util.Map;
 import java.util.Set;
-
-import static uk.gov.hmcts.ccd.domain.service.common.AccessControlService.*;
+import static uk.gov.hmcts.ccd.domain.service.common.AccessControlService.CAN_CREATE;
+import static uk.gov.hmcts.ccd.domain.service.common.AccessControlService.CAN_READ;
+import static uk.gov.hmcts.ccd.domain.service.common.AccessControlService.NO_CASE_TYPE_FOUND;
```
Comment (mario-paniccia) [inline src/main/java/uk/gov/hmcts/ccd/domain/service/createcase/AuthorisedCreateCaseOperation.java]: can't recall if we have a v1 and v2 endpoint for create case submission. Can we check and make sure we are covering both the endpoints?

Comment (kiran-yenigala-hmcts) [reply src/main/java/uk/gov/hmcts/ccd/domain/service/createcase/AuthorisedCreateCaseOperation.java]: Need to create IT for uk.gov.hmcts.ccd.v2.external.controller.CaseController

File: src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseDetailsEndpointIT.java [hunk 5]
```diff
@@ -3992,6 +4021,122 @@ public void shouldReturn200_whenSearchParamsAreSanitizedAsCaseWorker() throws Ex
         assertThat(responseAsString, containsString("Peter"));
     }
 
+    @Test
+    public void shouldReturn201WhenPostCreateCaseWithCreatorRoleWithNoDataForCaseworker() throws Exception {
```
Comment (mario-paniccia) [inline src/test/java/uk/gov/hmcts/ccd/endpoint/std/CaseDetailsEndpointIT.java]: shouldn't we move these tests to CaseControllerTestIT.java ?

**Pull request closed without merging**